modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/helper/AddFormInstanceRecordMVCCommandHelper.java

```diff
@@ -262,36 +262,42 @@ private void _removeDDMFormFieldValues(
 		Map<String, List<DDMFormFieldValue>> ddmFormFieldValuesMap,
 		Set<String> nonevaluableFieldNames) {
 
-		Stream<String> stream = nonevaluableFieldNames.stream();
-
-		stream.map(
-			ddmFormFieldValuesMap::get
-		).flatMap(
-			List::stream
-		).filter(
-			ddmFormFieldValue -> ddmFormFieldValue.getValue() != null
-		).forEach(
-			this::_removeDDMFormFieldValue
-		);
+		List<DDMFormFieldValue> ddmFormFieldValues = new ArrayList<>();
+
+		for (String nonevaluableFieldName : nonevaluableFieldNames) {
+			ddmFormFieldValues.addAll(
+				ddmFormFieldValuesMap.get(nonevaluableFieldName));
+		}
+
+		TransformUtil.transform(
```

Comment:
@LisztLiszt Should not use TransformUtil here, because the original logic is not collecting anything into a list. Just do a 2-layer for-loop (out for is the for-loop at line 267).